### ISSUE TITLE
[Random Randomizer] Sanctuary Key chest in Magma Chamber is randomized

### ISSUE THREAD

**Issue by Eradev** [Eradev/MonsterSanctuaryMods#3]
Sanctuary Key chest in Magma Chamber is randomized when it shouldn't.

(Reported by: Durandal#7401)

**Comment by Eradev** [comment]
Fixed in 1.1.3.

**Comment by garfieldbanks** [comment]
Is 1.1.3 available for download anywhere? The changelog on modworkshop.net shows the latest version as 1.1.2.

**Comment by Eradev** [comment]
No. I wanted to fix the others keys before releasing it, but I never got to do them.

**Comment by garfieldbanks** [comment]
Is the game still finishable using this randomizer? Just harder to find the keys? Did you commit any changes for the fix you made or is that code private? What would be required to add the rest?

**Comment by Eradev** [comment]
You need to disable the mod in certain maps before taking the content of the chests but otherwise you can.

I haven't comitted the code yet. I need to get to the bugged chests and check their IDs to blacklist them, but I haven't played in a long time now. If you can provide a save file before the bugged chest(s) then it would be welcome.

**Comment by garfieldbanks** [comment]
OK I was planning on making those saves you asked about but first I decided to try compiling the code and I get these errors. I'm guessing I'm missing a nuget package?
![err1](https://user-images.githubusercontent.com/12904270/229526410-9c07ee0c-5889-4ce5-879d-6f3e608a36cb.png)
![err2](https://user-images.githubusercontent.com/12904270/229526454-adf52c27-8f88-4922-bf37-3b0591eebcd4.png)

Any idea which packages I should install? I already installed everything that looked like it might be needed from https://nuget.bepinex.dev/

**Comment by garfieldbanks** [comment]
OK I think I actually solved that by installing Lib.Harmony. Now I have this error:
![err3](https://user-images.githubusercontent.com/12904270/229528449-7322a029-a6fe-4321-979c-416cba04e482.png)

**Comment by Eradev** [comment]
It's a BepInEx 5 plugin in Mono, and not IL2CPP.

**Comment by garfieldbanks** [comment]
Where do I get the bepinex 5 plugin mono version from?

**Comment by garfieldbanks** [comment]
OK I found them but they are prerelease version 6 and your project requires 5. Where do I get the nuget packages for the version 5?

**Comment by Eradev** [comment]
You can go to the `Tools -> NuGet Package Manager -> Manage Packges for Solution` and select which versions to install.

**Comment by garfieldbanks** [comment]
All I see is a long list of 6.0 preleases. Is there another nuget source I need to add?

**Comment by garfieldbanks** [comment]
I tried just adding the DLLs from the bepinex 5 release. But it seems to only want to check the boxes for three of them (I guess it only needs those). And it still has problems finding NameMenu and SFXController.
![err4](https://user-images.githubusercontent.com/12904270/229542231-f6572b43-b700-4c81-a24e-295e33d937b0.png)

**Comment by garfieldbanks** [comment]
Do I need to install FMOD studio or something for SFXController? What is NameMenu supposed to be coming from?

**Comment by Eradev** [comment]
I don't have time to debug your installation. Please follow the instructions on https://docs.bepinex.dev/articles/dev_guide/plugin_tutorial/index.html to setup your environment. Then get a fresh version of the code, and just restore the NuGet packages instead of installing random packages.

You need to add the reference to the game's Assembly-CSharp.dll for all mods, and possibly other of the game's DLL for certain ones.

**Comment by garfieldbanks** [comment]
OK. I got it to compile. I started over with a fresh git clone and retargeted it all to dotnet 4.8. Then I changed the hint references from what you had set in your path to the path to my assembly-csharp dll. I did the same thing with the post build event to point to the correct folder.

**Comment by garfieldbanks** [comment]
I added a bit to the code and compiled it:
```
if (__instance.Item == null && __instance.Gold == 0)
{
    _log.LogDebug("ChestRandomizer ignore: null and zero gold");

    return;
}
```
Any reason why you think this wouldn't fix the key chest issue?
